### PR TITLE
Turn last validation warnings into errors

### PR DESCRIPTION
**Pull request by melix**
This PR turns the remaining validation warnings into errors (except those which are for 8.0).

### CODE REVIEW

**Comment by melix** [comment]
@bot-gradle test QFL

**Comment by melix** [comment]
@bot-gradle test RFM

File: subprojects/docs/src/docs/userguide/troubleshooting/validation_problems.adoc [hunk 1]
```diff
@@ -324,3 +327,15 @@ If you didn't intend to write in this directory, you should simply setup your ta
 This error indicates that a collection of files, or a nested collection of files, contains an element that Gradle cannot convert to a file.
 
 To fix this, look at the error message which indicates the list of supported file notations and make sure to pick one of them.
+
+[[cannot_use_optional_on_primitive_types]]
+== Invalid use @Optional annotation on primitive types
```
Comment (wolfs) [inline subprojects/docs/src/docs/userguide/troubleshooting/validation_problems.adoc:332]:
```suggestion
== Invalid use of @Optional annotation on primitive types
```

**Review by wolfs — COMMENTED** [review]
LGTM! I left a small comment.

**Comment by melix** [comment]
@bot-gradle test RFM

**Comment by melix** [comment]
@bot-gradle test RFM

**Comment by melix** [comment]
@bot-gradle test RFM

**Pull request merged**